src/clear_cmd.cpp

```diff
@@ -365,7 +365,7 @@ void GenerateClearTile()
 
 		IncreaseGeneratingWorldProgress(GWP_ROUGH_ROCKY);
 		if (IsTileType(tile, MP_CLEAR) && !IsClearGround(tile, CLEAR_DESERT)) {
-			uint j = GB(r, 16, 4) + 5;
+			uint j = GB(r, 16, 4) + _settings_game.game_creation.amount_of_rocks;
```

Comment:
At the moment the range of j is 5 - 20.
The setting adjusts that from 1 - 16, to 255 - 270.
Without having actually tested it yet, it seems like it would be good to preserve the dynamic range in a logarithmic sense.
Perhaps the fixed 4 bits should be a dynamic value instead.

Reply:
> 
> 
> At the moment the range of j is 5 - 20.
> The setting adjusts that from 1 - 16, to 255 - 270.
> Without having actually tested it yet, it seems like it would be good to preserve the dynamic range in a logarithmic sense.
> Perhaps the fixed 4 bits should be a dynamic value instead.

So I had a play with this and it ended up causing some bizarre results; touching that fixed value of 4 rapidly resulted in a map completely filled with rocks, changing that sixteen to allow higher values in the range caused landscape generation to blow out entirely and result in tens of thousands of 'steps' to complete.

The best result came from keeping how it is and pushing the 'floor' of the range upwards, it certainly behaved more 'predictably' from a user standpoint.

I'm going to keep playing with different values but so-far leaving the overall size of the range alone, and pushing the floor up by either a fixed value or, something I've just added today pushing it up by the tile height value as well has resulted in better more predictable outcomes.

I'll update this PR with the changes for tile height shortly.